### DISCUSSION TITLE
XSSI prefix in JSON api response

### DISCUSSION THREAD

**Discussion by sleungcy** [grafana/grafana-infinity-datasource#926] [Q&A]
Hi all,

I am looking at an issue where the JSON api I am trying to query from infinity has some json magic prefix `([{` on the actual json object itself. According to their [documentation](https://security.stackexchange.com/questions/110539/how-does-including-a-magic-prefix-to-a-json-response-work-to-prevent-xssi-attack), the prefix is there to prevent XSSI exploitations. Is there anyway to configure infinity to strip out the prefix prior to parsing the data as JSON?

Thank in advance,
Sam

**Comment by YangKangSung** [comment]
I am facing the same issue. The response received from the Gerrit API includes a magic prefix, and it seems I need to remove the prefix first to enable JSON parsing. I would appreciate it if you could share a solution for this.

**Reply by YangKangSung** [reply]
I resolved the issue on my own. It seems that the only solution is backend processing. I used Flask or FastAPI to mediate between the source and the REST API, removing the "magic word" from the response before sending it back to the source.
